Build problem with MinGW (filesystem error)

### What happened?

Change https://github.com/gnuradio/gnuradio/commit/5173f01788a7e3c95e07286dbfca22e8b75d5273 caused an error when compiling with MinGW gcc.
Before this everything was fine.

### System Information

Host OS: Linux Ubuntu based 20.04 amd64
Compiler: GCC 11.4, mingw-w64 11.0.1 (custom toolchain)
GR Installation Method: Cross-compiling


### GNU Radio Version

3.11-git (main)

### Specific Version

v3.11.0.0git-668-gc2d1b5c5

### Steps to Reproduce the Problem

In `gr-fft/lib/fft.cc` I replaced the code blocks wrapped in `_MSC_VER` with `_WIN32`, and the build was successful.
But when trying to use Gnuradio it results in an error:
`filesystem error: cannot create directory: File exists [C:]`

### Relevant log output

```shell
terminate reached from thread id: 124Got std::system_error
filesystem error: cannot create directory: File exists [C:]
```

My MinGW toolchain is a highly modified [MXE](https://github.com/mxe/mxe)
<details>
  <summary>Build log without "fixes": </summary>
[ 38%] Building CXX object gr-fft/lib/CMakeFiles/gnuradio-fft.dir/fft.cc.obj
/media/buildsystem/gnuradio-main/gr-fft/lib/fft.cc: In function 'void gr::fft::wisdom_lock_init()':
/media/buildsystem/gnuradio-main/gr-fft/lib/fft.cc:78:36: error: cannot convert 'const value_type*' {aka 'const wchar_t*'} to 'const char*'
   78 |         open(wisdom_lock_file.c_str(), O_WRONLY | O_CREAT | O_NOCTTY | O_NONBLOCK, 0666);
      |              ~~~~~~~~~~~~~~~~~~~~~~^~
      |                                    |
      |                                    const value_type* {aka const wchar_t*}
In file included from /media/buildsystem/MinGW-M/usr/x86_64-w64-mingw32.shared/include/fcntl.h:8,
                 from /media/buildsystem/gnuradio-main/gr-fft/lib/fft.cc:29:
/media/buildsystem/MinGW-M/usr/x86_64-w64-mingw32.shared/include/io.h:340:32: note:   initializing argument 1 of 'int open(const char*, int, ...)'
  340 |   int __cdecl open(const char *_Filename,int _OpenFlag,...)  __MINGW_ATTRIB_DEPRECATED_MSVC2005;
      |                    ~~~~~~~~~~~~^~~~~~~~~
/media/buildsystem/gnuradio-main/gr-fft/lib/fft.cc: In function 'void gr::fft::import_wisdom()':
/media/buildsystem/gnuradio-main/gr-fft/lib/fft.cc:109:39: error: cannot convert 'const value_type*' {aka 'const wchar_t*'} to 'const char*'
  109 |     FILE* fp = fopen(wisdom_path.c_str(), "r");
      |                      ~~~~~~~~~~~~~~~~~^~
      |                                       |
      |                                       const value_type* {aka const wchar_t*}
In file included from /media/buildsystem/MinGW-M/usr/lib/gcc/x86_64-w64-mingw32.shared/11.4.0/include/c++/cstdio:42,
                 from /media/buildsystem/MinGW-M/usr/x86_64-w64-mingw32.shared/include/spdlog/fmt/bundled/core.h:12,
                 from /media/buildsystem/MinGW-M/usr/x86_64-w64-mingw32.shared/include/spdlog/fmt/fmt.h:27,
                 from /media/buildsystem/gnuradio-main/gnuradio-runtime/lib/../include/gnuradio/io_signature.h:19,
                 from /media/buildsystem/gnuradio-main/gnuradio-runtime/lib/../include/gnuradio/basic_block.h:15,
                 from /media/buildsystem/gnuradio-main/gnuradio-runtime/lib/../include/gnuradio/hier_block2.h:15,
                 from /media/buildsystem/gnuradio-main/gnuradio-runtime/include/gnuradio/top_block.h:15,
                 from /media/buildsystem/gnuradio-main/build/gr-fft/lib/CMakeFiles/gnuradio-fft.dir/cmake_pch.hxx:5,
                 from <command-line>:
/media/buildsystem/MinGW-M/usr/x86_64-w64-mingw32.shared/include/stdio.h:639:49: note:   initializing argument 1 of 'FILE* fopen(const char*, const char*)'
  639 |   FILE *__cdecl fopen(const char * __restrict__ _Filename,const char * __restrict__ _Mode) __MINGW_ATTRIB_DEPRECATED_SEC_WARN;
      |                       ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
/media/buildsystem/gnuradio-main/gr-fft/lib/fft.cc: In function 'void gr::fft::config_threading(int)':
/media/buildsystem/gnuradio-main/gr-fft/lib/fft.cc:123:16: warning: unused variable 'fftw_threads_inited' [-Wunused-variable]
  123 |     static int fftw_threads_inited = 0;
      |                ^~~~~~~~~~~~~~~~~~~
/media/buildsystem/gnuradio-main/gr-fft/lib/fft.cc: In function 'void gr::fft::export_wisdom()':
/media/buildsystem/gnuradio-main/gr-fft/lib/fft.cc:141:39: error: cannot convert 'const value_type*' {aka 'const wchar_t*'} to 'const char*'
  141 |     FILE* fp = fopen(wisdom_path.c_str(), "w");
      |                      ~~~~~~~~~~~~~~~~~^~
      |                                       |
      |                                       const value_type* {aka const wchar_t*}
In file included from /media/buildsystem/MinGW-M/usr/lib/gcc/x86_64-w64-mingw32.shared/11.4.0/include/c++/cstdio:42,
                 from /media/buildsystem/MinGW-M/usr/x86_64-w64-mingw32.shared/include/spdlog/fmt/bundled/core.h:12,
                 from /media/buildsystem/MinGW-M/usr/x86_64-w64-mingw32.shared/include/spdlog/fmt/fmt.h:27,
                 from /media/buildsystem/gnuradio-main/gnuradio-runtime/lib/../include/gnuradio/io_signature.h:19,
                 from /media/buildsystem/gnuradio-main/gnuradio-runtime/lib/../include/gnuradio/basic_block.h:15,
                 from /media/buildsystem/gnuradio-main/gnuradio-runtime/lib/../include/gnuradio/hier_block2.h:15,
                 from /media/buildsystem/gnuradio-main/gnuradio-runtime/include/gnuradio/top_block.h:15,
                 from /media/buildsystem/gnuradio-main/build/gr-fft/lib/CMakeFiles/gnuradio-fft.dir/cmake_pch.hxx:5,
                 from <command-line>:
/media/buildsystem/MinGW-M/usr/x86_64-w64-mingw32.shared/include/stdio.h:639:49: note:   initializing argument 1 of 'FILE* fopen(const char*, const char*)'
  639 |   FILE *__cdecl fopen(const char * __restrict__ _Filename,const char * __restrict__ _Mode) __MINGW_ATTRIB_DEPRECATED_SEC_WARN;
      |                       ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
/media/buildsystem/gnuradio-main/gr-fft/lib/fft.cc: At global scope:
/media/buildsystem/gnuradio-main/gr-fft/lib/fft.cc:123:16: warning: 'fftw_threads_inited' defined but not used [-Wunused-variable]
  123 |     static int fftw_threads_inited = 0;
      |                ^~~~~~~~~~~~~~~~~~~
make[2]: *** [gr-fft/lib/CMakeFiles/gnuradio-fft.dir/build.make:95: gr-fft/lib/CMakeFiles/gnuradio-fft.dir/fft.cc.obj] Error 1
make[1]: *** [CMakeFiles/Makefile2:2001: gr-fft/lib/CMakeFiles/gnuradio-fft.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
</details>

--- a/gnuradio-runtime/lib/vmcircbuf_prefs.cc
+++ b/gnuradio-runtime/lib/vmcircbuf_prefs.cc
@@ -37,17 +37,8 @@ static auto pathname(string_type key)
 static void ensure_dir_path()
 {
     // recursively make sure the directory exists
-    fs::path path;
-    for (const auto& path_component : gr::paths::userconf()) {
-        path /= path_component;
-        if (!fs::exists(path)) {
-            fs::create_directory(path);
-        }
-    }
-
-    path = path / "prefs";
-    if (!fs::exists(path))
-        fs::create_directory(path);
+    fs::path path = gr::paths::userconf() / "prefs";
+    fs::create_directories(path);
 }
 
 template <typename stream_t>

--- a/gr-fft/lib/fft.cc
+++ b/gr-fft/lib/fft.cc
@@ -62,22 +62,12 @@ static void wisdom_lock_init()
         return;
 
     // recursively make sure the directory exists
-    fs::path path;
-    for (const auto& path_component : gr::paths::cache()) {
-        path /= path_component;
-        if (!fs::exists(path)) {
-            fs::create_directory(path);
-        }
-    }
+    fs::path path = gr::paths::cache();
+    fs::create_directories(path);
     const auto wisdom_lock_file = path / WISDOM_LOCKFILE;
-#ifdef _MSC_VER
     int fd = open(wisdom_lock_file.string().c_str(),
                   O_WRONLY | O_CREAT | O_NOCTTY | O_NONBLOCK,
                   0666);
-#else
-    int fd =
-        open(wisdom_lock_file.c_str(), O_WRONLY | O_CREAT | O_NOCTTY | O_NONBLOCK, 0666);
-#endif
     if (fd < 0) {
         throw std::runtime_error("Failed to create FFTW wisdom lockfile: " +
                                  wisdom_lock_file.string());
@@ -104,11 +94,7 @@ static void unlock_wisdom()
 static void import_wisdom()
 {
     auto wisdom_path = gr::paths::cache() / WISDOM_FILENAME;
-#ifdef _MSC_VER
     FILE* fp = fopen(wisdom_path.string().c_str(), "r");
-#else
-    FILE* fp = fopen(wisdom_path.c_str(), "r");
-#endif
     if (fp != 0) {
         int r = fftwf_import_wisdom_from_file(fp);
         fclose(fp);
@@ -136,11 +122,7 @@ static void config_threading(int nthreads)
 static void export_wisdom()
 {
     auto wisdom_path = gr::paths::cache() / WISDOM_FILENAME;
-#ifdef _MSC_VER
     FILE* fp = fopen(wisdom_path.string().c_str(), "w");
-#else
-    FILE* fp = fopen(wisdom_path.c_str(), "w");
-#endif
     if (fp != 0) {
         fftwf_export_wisdom_to_file(fp);
         fclose(fp);

